examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,272 @@
+regularization discourages any single coefficient :math:`\\beta_{i}` from taking
+an excessively large value and encourages smaller and more evenly distributed
+coefficients. Higher values of :math:`\\alpha` force the coefficients towards
+zero. However, an excessively high :math:`\\alpha` can result in an underfit
+model that fails to capture important patterns in the data.
+
+Therefore, the regularized loss function combines the prediction accuracy term
+and the penalty term. By adjusting the regularization strength, practitioners
+can fine-tune the degree of constraint imposed on the weights, training a model
+capable of generalizing well to unseen data while avoiding overfitting.
+
+For a comparison between Ridge and Huber Regularization see
+:doc:`~examples.linear_model.plot_huber_vs_ridge`
 """
 
 # Author: Kornel Kielczewski -- <kornel.k@plusnet.pl>
 
-import matplotlib.pyplot as plt
-import numpy as np
+# %%
+# Purpose of this example
+# -----------------------
+# For the purpose of showing how Ridge regularization works, we will create two
+# data sets: non-noisy and noisy. Then we will train them on a range of
+# regularization strengths (:math:`\alpha`) and plot how the trained
+# coefficients and the mean squared error between those and the original values
+# behave as functions of the regularization strength.
+
+# %%
+# Ridge Regularization on non-noisy data set
+# ------------------------------------------
 
+# %%
+# Creating non-noisy data set
+#############################
+# We make a toy data set, that's suitable to detect regression. Please note that
+# in this example the data is non-noisy, hence it is possible to extract the
+# exact coefficients.
 from sklearn.datasets import make_regression
+
+X, y, w = make_regression(n_samples=100, n_features=10, coef=True, random_state=1)
+
+# Obtain the true coefficients
+print(f"The true coefficient of this regression problem are:\n{w}")
+
+# %%
+# Training Ridge Regressor
+##########################
+# We use :class:`~sklearn.linear_model.Ridge`, a linear model with L2
+# regularization. The model parameter `alpha` is a positive constant that
+# multiplies the penalty term, controlling the regularization strength.
+import numpy as np
+
 from sklearn.linear_model import Ridge
 from sklearn.metrics import mean_squared_error
 
 clf = Ridge()
 
+# Generate values for `alpha` that are evenly distributed on a logarithmic scale
+alphas = np.logspace(-3, 4, 200)
+coefs = []
+errors_coefs = []
+
+# Train the model with different regularisation strengths
+for a in alphas:
+    clf.set_params(alpha=a).fit(X, y)
+    coefs.append(clf.coef_)
+    errors_coefs.append(mean_squared_error(clf.coef_, w))
+
+# %%
+# Note: In a real setting you should use :class:`~sklearn.linear_model.RidgeCV`
+# for training a Ridge regression with built-in cross-validation, but for the
+# purpose of demonstration we will keep it simple here.
```

Comment:
Let's skip the note. We will keep it for when we have a nice example for the second section.